nsxt/resource_nsxt_upgrade_precheck_acknowledge.go

```diff
@@ -0,0 +1,138 @@
+/* Copyright © 2023 VMware, Inc. All Rights Reserved.
+   SPDX-License-Identifier: MPL-2.0 */
+
+package nsxt
+
+import (
+	"fmt"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	nsxModel "github.com/vmware/vsphere-automation-sdk-go/services/nsxt-mp/nsx/model"
+	"github.com/vmware/vsphere-automation-sdk-go/services/nsxt-mp/nsx/upgrade"
+)
+
+func resourceNsxtUpgradePrecheckAcknowledge() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceNsxtUpgradePrecheckAcknowledgeCreate,
+		Read:   resourceNsxtUpgradePrecheckAcknowledgeRead,
+		Update: resourceNsxtUpgradePrecheckAcknowledgeUpdate,
+		Delete: resourceNsxtUpgradePrecheckAcknowledgeDelete,
+
+		Schema: map[string]*schema.Schema{
+			"precheck_ids": {
+				Type:        schema.TypeList,
+				Description: "IDs of precheck warnings that need to be acknowledged",
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+				Required: true,
+			},
+			"precheck_warnings": {
+				Type:        schema.TypeList,
+				Description: "List of warnings from precheck",
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"id": {
+							Type:        schema.TypeString,
+							Description: "ID of the precheck warning",
+							Computed:    true,
+						},
+						"message": {
+							Type:        schema.TypeString,
+							Description: "Message of the precheck warning",
+							Computed:    true,
+						},
+						"is_acknowledged": {
+							Type:        schema.TypeBool,
+							Description: "Boolean value which identifies if the precheck warning has been acknowledged.",
+							Computed:    true,
+						},
+					},
+				},
+				Computed: true,
+			},
+		},
+	}
+}
+
+func resourceNsxtUpgradePrecheckAcknowledgeCreate(d *schema.ResourceData, m interface{}) error {
+	id := d.Id()
+	if id == "" {
+		id = newUUID()
+	}
+	d.SetId(id)
+	precheckIDs := interface2StringList(d.Get("precheck_ids").([]interface{}))
+	err := acknowledgePrecheckWarnings(m, precheckIDs)
+	if err != nil {
+		return handleCreateError("NsxtPrecheckAcknowledge", id, err)
+	}
+	return resourceNsxtUpgradePrecheckAcknowledgeRead(d, m)
+}
+
+func acknowledgePrecheckWarnings(m interface{}, precheckIDs []string) error {
+	connector := getPolicyConnector(m)
+	client := upgrade.NewPreUpgradeChecksClient(connector)
+	for _, precheckID := range precheckIDs {
+		err := client.Acknowledge(precheckID)
+		if err != nil {
+			return fmt.Errorf("Failed to acknowledge precheck warning with ID %s: %s", precheckID, err)
```

Comment:
If formatted error is returned here, `handleCreateError` will fail to parse it as NSX error, and the user will see something like ` Failed to acknowledge precheck warning with ID backupOperationCheck: com.vmware.vapi.std.errors.invalid_request`
I would suggest to either return `err` here, or use `logApiError` to convey NSX message to the user

Reply:
Done